From 94d2ddb00a2dbcaa4b352a3fceaaa5e6296a9ae0 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sun, 13 Nov 2016 02:01:45 -0500
Subject: [PATCH] force 32-bit build to fix garbled output

Pico TTS is currently broken on 64-bit. There was an attempt to port it,
but the voices are garbled, perhaps due to integer truncation. Until it
gets fixed (if ever), it needs to be compiled as 32-bit.

Change-Id: Iae009dafc41796169999cab2dae4b246be2ea0f4
---

diff --git a/pico/Android.mk b/pico/Android.mk
index 2f6420e..e35e6be 100755
--- a/pico/Android.mk
+++ b/pico/Android.mk
@@ -8,6 +8,7 @@
 include $(CLEAR_VARS)
 
 LOCAL_PACKAGE_NAME := PicoTts
+LOCAL_MULTILIB := 32
 
 LOCAL_SRC_FILES := \
     $(call all-java-files-under, src) \
diff --git a/pico/compat/jni/Android.mk b/pico/compat/jni/Android.mk
index 5a7180c..9b3ad60 100755
--- a/pico/compat/jni/Android.mk
+++ b/pico/compat/jni/Android.mk
@@ -2,6 +2,7 @@
 include $(CLEAR_VARS)
 
 LOCAL_MODULE:= libttscompat
+LOCAL_MULTILIB := 32
 
 LOCAL_SRC_FILES:= \
 	com_android_tts_compat_SynthProxy.cpp
diff --git a/pico/lib/Android.mk b/pico/lib/Android.mk
index 70f0099..cdf0026 100644
--- a/pico/lib/Android.mk
+++ b/pico/lib/Android.mk
@@ -4,6 +4,7 @@
 include $(CLEAR_VARS)
 
 LOCAL_MODULE := libsvoxpico
+LOCAL_MULTILIB := 32
 
 LOCAL_SRC_FILES := \
 	picoacph.c \
diff --git a/pico/tts/Android.mk b/pico/tts/Android.mk
index 335a1c2..0f0d737 100644
--- a/pico/tts/Android.mk
+++ b/pico/tts/Android.mk
@@ -7,6 +7,7 @@
 include $(CLEAR_VARS)
 
 LOCAL_MODULE := libttspico_engine
+LOCAL_MULTILIB := 32
 
 LOCAL_SRC_FILES := \
 	com_svox_picottsengine.cpp \
@@ -36,6 +37,7 @@
 include $(CLEAR_VARS)
 
 LOCAL_MODULE := libttspico
+LOCAL_MULTILIB := 32
 	
 LOCAL_SRC_FILES := \
 	com_svox_picottsengine.cpp \
